### ISSUE TITLE
Deprecation false-positive: Function.extractParams is deprecated

### ISSUE THREAD

**Issue by lierdakil** [atom/atom#15646]
### Description

`TextEditor.addCursorAtBufferPosition(pos, { autoscroll: false })` triggers the following deprecation warning:

> Assigning custom properties to a marker when creating/copying it is
deprecated. Please, consider storing the custom properties you need in
some other object in your package, keyed by the marker's id property.

This is a false positive, since `autoscroll` is not a marker property, but rather an option for `addCursorAtBufferPosition` itself.

I believe https://github.com/atom/atom/commit/b362f746f8fb21d9850f02ca82cc50e90e126d38 is at fault.

### Steps to Reproduce

1. run `atom.workspace.getActiveTextEditor().addCursorAtBufferPosition([0,0], { autoscroll: false })` in dev. console.
2. Open Deprecation Cop

**Expected behavior:** No deprecations

**Actual behavior:** 1 deprecation in atom core:
> atom core (1 deprecation)
> 
> Assigning custom properties to a marker when creating/copying it is deprecated. Please, consider storing the custom properties you need in some other object in your package, keyed by the marker's id property.
```
Function.extractParams - <embedded>:127825:20
MarkerLayer.markPosition - <embedded>:73406:30
DisplayMarkerLayer.markBufferPosition - <embedded>:128445:58
TextEditor.addCursorAtBufferPosition - <embedded>:4226:40
<unknown> - :1:38
```

**Reproduces how often:** 100%

### Versions

Tested on Atom 1.20. I believe all versions since 1.19.0-beta0 are affected.

### Additional Information

Initially reported at https://github.com/atom-haskell/ide-haskell/issues/212

Related: https://github.com/atom/text-buffer/pull/149 (PR that introduced deprecation message in question)

**Comment by rsese** [comment]
Thanks for the report!  I can repro the deprecation message in `1.22.0-dev-a77538ab8` - I'll ask the team to take a look.

**Comment by iolsen** [comment]
We should remove the deprecation warning and update the docs to include this property.

**Comment by maxbrunsfeld** [comment]
Thanks for the detailed report @lierdakil! This will be fixed in v1.22.
